presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -444,7 +444,11 @@ else if (!javaType.isPrimitive()) {
         @Override
         protected Object visitIdentifier(Identifier node, Object context)
         {
-            return node;
+            // Identifier only exists before planning.
+            // ExpressionInterpreter should only be invoked after planning.
+            // As a result, this method should be unreachable.
+            // However, RelationPlanner.visitUnnest and visitValues invokes evaluateConstantExpression.
+            return ((SymbolResolver) context).getValue(new Symbol(node.getValue()));
```

Comment:
is `context` always of type `SymbolResolver`?

Reply:
Maybe add an `instanceof` check here and return null or throw an internal error?

Reply:
No point. That would be a bug and it should fail in the same way every other checkState or assertion error does